### ISSUE TITLE
Will fine-tuning training functionality for SAM3 be added?

### ISSUE THREAD

**Issue by yang-0201** [ultralytics/ultralytics#24210]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Thank you for your work and contributions — I’ve benefited a lot from them. I would like to ask whether the team has any plans to release fine-tuning training support for SAM3 in the future.

The original SAM3 training pipeline and scripts are relatively complex and not very user-friendly. In contrast, the Ultralytics library already provides a more convenient dataset loading approach similar to YOLOE. If it were possible to build a SAM3-based training framework in a similar way, it could significantly enhance the application of open-set detection and segmentation models in specialized domains, which would be a very meaningful contribution.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yang-0201, thank you for your interest in Ultralytics 🚀 and for the thoughtful feature request around SAM3 fine-tuning support. This is an automated response to help get your issue routed quickly, and an Ultralytics engineer will also assist soon 😊

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question or feature request, please provide as much information as possible, including your target use case, dataset image examples, expected training workflow, and any relevant logs or prototype code, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To help the team evaluate this request faster, it would be especially useful if you could share:
- The specific SAM3 fine-tuning workflow you would like to see supported
- The type of dataset format and annotation pipeline you expect
- Whether your main goal is open-set detection, segmentation, or both
- Any examples of how you would want the Ultralytics-style training interface to look

Thanks again for sharing this idea 💡

**Comment by rehan243** [comment]
This is an interesting request. From our experience, integrating SAM-like models into a user-friendly training pipeline involves a few key considerations:

- The original SAM3 training scripts tend to be tightly coupled with their own data processing and augmentation routines, so wrapping them into Ultralytics-style datasets would require creating custom `Dataset` classes compatible with both models.
- Leveraging existing tools like `torchvision` or `albumentations` for data augmentation can make the pipeline more flexible, similar to how Ultralytics handles YOLO data.
- Using LoRA-based fine-tuning (which we've done for other models) could keep training efficient, especially given the large size of SAM3, allowing us to adapt it with fewer resources.
- While there’s no current official support, a starting point could be to build a lightweight wrapper around SAM3 inference and training scripts, integrating it into the Ultralytics framework for dataset loading and training loops.

If you’re up for a contribution, providing a minimal example of such a pipeline would be very valuable for the community.

**Comment by Y-T-G** [comment]
Thanks for the feature request. Fine-tuning feature for SAM3 is currently not in our roadmap. SAM3 training pipeline is very different from YOLO and YOLOE training pipeline. It will require significant work to implement it, and also significant work to maintain it afterwards. It might be considered only if there's significant number of users asking for it.

**Comment by ZLT-a-LL** [comment]
For some open-source projects that convert sam3 quantization to TensorRT with fp16, although they are nearly twice as fast as PT models, they still struggle to meet real-time requirements on some edge devices. I would like to ask if Ultralytics has plans to perform int8 quantization fine-tuning on sam3 to make its application more widespread

**Comment by glenn-jocher** [comment]
Thanks for raising this; there’s no announced Ultralytics support for SAM3 INT8 quantization-aware fine-tuning at the moment, and the current [SAM 3 docs](https://docs.ultralytics.com/models/sam-3/) cover inference rather than training/quantization, so please open a separate feature request for INT8/TensorRT optimization to track demand clearly.
